api/api.go

```diff
@@ -0,0 +1,527 @@
+	if err := json.Unmarshal(resp, &secretMap); err != nil {
+		return nil, fmt.Errorf("unmarshalling Coordinator response: %w", err)
+	}
+	return secretMap, nil
+}
+
+// SecretSet sets secrets for a MarbleRun deployment.
+func SecretSet(ctx context.Context, endpoint string, trustedRoot *x509.Certificate, clientKeyPair *tls.Certificate, secrets map[string]manifest.UserSecret) error {
+	client, err := rest.NewClient(endpoint, trustedRoot, clientKeyPair)
+	if err != nil {
+		return fmt.Errorf("setting up client: %w", err)
+	}
+
+	secretDataJSON, err := json.Marshal(secrets)
+	if err != nil {
+		return fmt.Errorf("marshalling secrets: %w", err)
+	}
+	_, err = client.Post(ctx, rest.SecretEndpoint, rest.ContentJSON, bytes.NewReader(secretDataJSON))
+	return err
+}
+
+// VerifyOptions specifies how to verify the remote attestation of a Coordinator instances.
+type VerifyOptions struct {
+	// InsecureSkipVerify disables verification of the Coordinator's attestation statement.
+	// WARNING: This IS insecure and should only be used for testing purposes.
+	InsecureSkipVerify bool `json:"InsecureSkipVerify"`
+
+	// UniqueID is the unique ID (MRENCLAVE) of the Coordinator enclave.
+	UniqueID string `json:"UniqueID"`
+	// SignerID is the signer ID (MRSIGNER) of the Coordinator enclave.
+	SignerID string `json:"SignerID"`
+	// SecurityVersion is the security version (ISVSVN) of the Coordinator enclave.
+	SecurityVersion uint `json:"SecurityVersion"`
+	// ProductID is the product ID (ISVPRODID) of the Coordinator enclave.
+	ProductID uint16 `json:"ProductID"`
+	// Debug specifies whether the Coordinator enclave is allowed to run in debug mode.
+	Debug bool `json:"Debug"`
+
+	// AcceptedTCBStatuses is a list of TCB statuses that are considered valid.
+	// Should be one or multiple from {"UpToDate", "OutOfDate",	"Revoked", "ConfigurationNeeded", "OutOfDateConfigurationNeeded", "SWHardeningNeeded", "ConfigurationAndSWHardeningNeeded"}.
+	// If not set, defaults to ["UpToDate", "SWHardeningNeeded"].
+	// If the Coordinator returns a TCB status not listed, an [attestation.TCBStatusError] is returned.
+	AcceptedTCBStatuses []string `json:"AcceptedTCBStatuses"`
+
+	// Nonce is an optional, user-defined nonce to be included in the Coordinator's attestation statement.
+	// If set, the Coordinator will generate an SGX quote over sha256(Coordinator_root_cert, Nonce).
+	Nonce []byte `json:"Nonce"`
+}
+
+// VerifyOptionsFromConfig reads a configuration file from disk.
+func VerifyOptionsFromConfig(configPath string) (VerifyOptions, error) {
+	fs := afero.NewOsFs()
+	return verifyOptionsFromConfig(afero.Afero{Fs: fs}, configPath)
+}
+
+func (v *VerifyOptions) setDefaults() {
+	if len(v.AcceptedTCBStatuses) == 0 {
+		v.AcceptedTCBStatuses = []string{"UpToDate", "SWHardeningNeeded"}
+	}
+}
+
+func verifyOptionsFromConfig(fs afero.Afero, configPath string) (VerifyOptions, error) {
+	var opts VerifyOptions
+	optsRaw, err := fs.ReadFile(configPath)
```

Comment:
did you plan to write unit tests, or what is the split func and use of afero about?

Reply:
Yes, though I'm currently unsure how worthwhile they are for the functions here, which mostly just unmarshal and return a value.
I feel this is better handled by integration or e2e tests.
We may add unit tests for completeness nonetheless.

Reply:
I agree that unit tests don't add much value here. We should better focus on integration/e2e and make sure that those cover all of the API.

Reply:
Our e2e tests already cover usage of the API calls through the CLI, added a ticket to update the integration test: 
https://dev.azure.com/Edgeless/Edgeless/_workitems/edit/4185